FF1Lib/asm/1E.asm

```diff
@@ -164,66 +185,61 @@ DoPartyGen_OnCharacter:
         ; if B pressed -- just SEC and exit
         SEC
         RTS
-      
+
       ; Code reaches here if A/B were not pressed
     : LDA joy
       AND #$0F
       CMP joy_prevdir
       BEQ @MainLoop             ; if there was no change in directional input, loop to another frame
-      
+
       STA joy_prevdir           ; otherwise, record new directional input as prevdir
       CMP #$00                  ; if directional input released (rather than pressed)
       BEQ @MainLoop             ;   loop to another frame.
-    
+
      ; Otherwise, if any direction was pressed:
-   
-      CLC
-      retry:
-        ROR tmp3
-        BCC skip
-          ROR tmp3
-          ROR tmp3
-        skip LDX char_index
-        INC ptyclass,X
-        LDA ptyclass,X
-        SBC #$07
-        CMP #$FF
-        BNE skip2
-          STA ptyclass,X
-        skip2:
-        LDA tmp
-        BIT tmp3
-        BEQ retry
 
+      LDX char_index
+     @retry:
+        LDA ptygen_class,X         ; A = class id.  0=FI, 1=TH, BB, RM, WM, BM, FF=None
+        CLC
+        ADC #1
+        CMP #6
+        BNE :+
+          LDA #$FF
+      : STA ptygen_class, X
+        TAY
+        LDA lut_ClassMask,Y

```

Comment:
Y can be 0-5 inclusive, and FF, if Y is FF, it does not load the correct value, for example from the fceux debugger
`1E:80FF:B9 15 81  LDA $8115,Y @ $8214 = #$83`
this should be loading address $8114

Reply:
I did test this, but maybe I changed it during testing and checked in the wrong stuff. It was many weeks ago.

So what is supposed to be going on is that Y==FF results in LDA lut_ClassMask-1, but obviously that isn't happening. This only works if lut_ClassMask is on the zero page, and then you have to use op B5 instead of op B9.

Anyway, this is fixable with adding INY between TAY and LDA, and then moving all the lut items up one. Less cool though.